netbox_ip_status.py

```diff
@@ -17,110 +15,74 @@
 )
 nb.http_session = netbox_session
 
+# 更新対象の prefix を取得
 prefixes = nb.ipam.prefixes.filter(tag=[config.PREFIX_TAG])
 
 today_datetime = datetime.datetime.now()
 today = today_datetime.strftime('%Y-%m-%d')
 
-last_seen = {}
-
-def update_addresses(addresses, prefix_mask):
-    for address in addresses:
-        update_address(address, prefix_mask)
-
+# IP アドレスからホスト名を逆引き
 def reverse_lookup(ip):
     try:
         hostname, _, _ = socket.gethostbyaddr(ip)
         return hostname
     except socket.herror:
         return None
 
-def update_address(ipy_address, prefix_mask):
+# IP アドレスのステータスを更新
+def update_addresses(addresses, prefix_mask):
+    for address in addresses:
+        update_address(address, prefix_mask)
+
+# IP アドレスのステータスを更新
+# TODO: 状態チェックとNetboxへの登録は別々に行う
+def update_address(ipy_address, prefix_mask = "24"):
+    print(ipy_address.strNormal() + '/' + str(prefix_mask))
+
     ip = ipy_address.strNormal()
     updated = False
     try:
         ping_result = ping(address=ip, timeout=0.5, interval=1, count=3)
         rev = reverse_lookup(ip)
         address = nb.ipam.ip_addresses.get(address=ipy_address.strNormal(1))
-        if address is not None:
-            new_tag = None
 
+        if address is not None:
             if ping_result.is_alive:
-                last_seen[str(address)] = today
-                new_tag = {"name": "lastseen:today"}
-            else:
-                if str(address) in last_seen.keys():
-                    last_seen_date = datetime.datetime.strptime(last_seen[str(address)], '%Y-%m-%d')
-                    #lastseen = datetime.datetime.strptime("2021-01-01", '%Y-%m-%d')
-                    delta = today_datetime - last_seen_date
-                    delta = delta.days
-                    if delta == 0:
-                        new_tag = {"name": "lastseen:today"}
-                    elif delta == 1:
-                        new_tag = {"name": "lastseen:yesterday"}
-                    elif delta > 1 and delta < 8:
-                        new_tag = {"name": "lastseen:week"}
-                    elif delta > 7 and delta < 32:
-                        new_tag = {"name": "lastseen:month"}
-                    elif delta > 31 and delta < 366:
-                        new_tag = {"name": "lastseen:year"}
-                    else:
-                        new_tag = {"name": "lastseen:overayear"}
-                else:
-                    # Last seen is none, and we haven't been able to see it today either!
-                    new_tag = {"name": "lastseen:never"}
-            
-            # Only update reverse DNS if it changes
-            if rev is not None:
-                if address.dns_name != rev:
-                    address.dns_name = rev
+                # MEMO: deprecated / reserved の時に ping が通るようになった場合、status を active に戻す
+                if address.status in ['deprecated', 'reserved']:
```

Comment:
細かいけど、listじゃなくてsetのほうがいいかも？
```suggestion
                if address.status in {'deprecated', 'reserved'}:
```

Reply:
@tadaren というと...?

Reply:
@NAKNAO-nnct 
`in`で要素検索するときに、listだと線形探索だけどsetならhashで検索するからいいかなって話だけど、要素数2でそこまで気にする必要もないので修正はどっちでもいい。修正面倒なら放置でもOK

Reply:
今まで そんなこと考えたことなかったわ。
気が向いたら治しておく

Reply:
[81f1422](https://github.com/Yamada-Factory/netbox-ip-status/pull/1/commits/81f14224024a81e741fb55a5526f20adea36df61)